xcode/Subconscious/Shared/Models/Slashlink.swift

```diff
@@ -67,11 +67,27 @@ public struct Slashlink:
         else {
             return nil
         }
-        let slug = Slug(uncheckedRawString: match.slug.toString())
+        
+        // There are four cases: petname-only, slug-only, petname+slug and empty.
+        // All are valid constructions except for empty.
+        // Petname-only will use `profileSlug` as the slug.
+        let slug = match.slug.map({ substring in
+            Slug(uncheckedRawString: substring.toString()
+        )})
         let petname = match.petname.map({ substring in
             Petname(uncheckedRawString: substring.toString())
         })
-        self.init(petname: petname, slug: slug)
+        
+        if let slug = slug {
+            self.init(petname: petname, slug: slug)
+        } else {
+            guard let petname = petname else {
+                return nil
+            }
+            
+            // Petname only
+            self.init(petname: petname, slug: Self.profileSlug)
+        }
```

Comment:
I think we may be able to restructure this logic with a tuple and switch statement:

```swift
switch (petname, slug) {
    // ...
}
```

which would be nice because then we get compiler-checked for exhaustive matching.

Reply:
I was looking for syntax like this, TIL! I didn't know about the unwrapping syntax using `.some(let x)`